lib/elixir/src/elixir_expand.erl

```diff
@@ -27,6 +27,11 @@ expand({'%', Meta, [Left, Right]}, E) ->
 expand({'<<>>', Meta, Args}, E) ->
   elixir_bitstring:expand(Meta, Args, E, false);
 
+expand({'module_sigil', Meta, [Module, String, Modifiers]}, E) ->
+  Alias = {'__aliases__', [{alias, false}], [Module]},
+  RE = E#{requires := ordsets:add_element(Module, ?key(E, requires))},
```

Comment:
the consequence of this is:

```elixir
iex(1)> __ENV__.requires
[Application, IEx.Helpers, Kernel, Kernel.Typespec]
iex(2)> ~URI[https://elixir-lang.org]
iex(3)> __ENV__.requires
[Application, IEx.Helpers, Kernel, Kernel.Typespec, URI]
```

which might be surprising. At the same time, without automatically requiring the module, the feature feels less ergonomic to use. (But it would match the behaviour of "regular" sigils, if it's a macro that module needs to be required.)